src/ring_theory/polynomial/cyclotomic.lean

```diff
@@ -451,6 +451,77 @@ begin
   cyclotomic'_eq_X_pow_sub_one_div hpos hz, finset.prod_congr (refl k.proper_divisors) h]
 end
 
+lemma eq_cyclotomic_iff {R : Type*} [comm_ring R] [nontrivial R] {n : ℕ} (hpos: 0 < n)
+  (P : polynomial R) :
+  P = cyclotomic n R ↔ P * (∏ i in nat.proper_divisors n, polynomial.cyclotomic i R) = X ^ n - 1 :=
+begin
+  split,
+  { intro hcycl,
+    rw [hcycl, ←finset.prod_insert (@nat.proper_divisors.not_self_mem n),
+    ←nat.divisors_eq_proper_divisors_insert_self_of_pos hpos],
+    exact prod_cyclotomic_eq_X_pow_sub_one hpos R },
+  { intro hP,
+    have prod_monic : (∏ i in nat.proper_divisors n, cyclotomic i R).monic,
+    { apply monic_prod_of_monic,
+      intros i hi,
+      exact cyclotomic.monic i R },
+    rw [@cyclotomic_eq_X_pow_sub_one_div R _ _ _ hpos,
+    (div_mod_by_monic_unique P 0 prod_monic _).1],
+    split,
+    { rwa [zero_add, mul_comm] },
+    rw [degree_zero, bot_lt_iff_ne_bot],
+    intro h,
+    exact monic.ne_zero prod_monic (degree_eq_bot.1 h) },
+end
+
+/-- If `p` is prime, then `cyclotomic p R = geom_series X p`. -/
+lemma cyclotomic_of_prime {R : Type*} [comm_ring R] [nontrivial R] {p : ℕ} (hp : nat.prime p) :
+  geom_series X p = cyclotomic p R :=
+begin
+  apply (@eq_cyclotomic_iff R _ _ p (nat.prime.pos hp) _).2,
+  rw nat.prime.proper_divisors hp,
+  simp only [geom_sum_mul, finset.prod_singleton, cyclotomic_one],
+end
+
+/-- The constant term of `cyclotomic n R` is `1` if `2 ≤ n`. -/
+lemma cyclotomic_coeff_zero {R : Type*} [comm_ring R] (n : ℕ) (hn : 2 ≤ n) :
+  (cyclotomic n R).coeff 0 = 1 :=
+begin
+  revert hn,
+  apply nat.strong_induction_on n,
+  intros k hi hk,
+  have hprod : (∏ i in nat.proper_divisors k, (polynomial.cyclotomic i R).coeff 0) = -1,
+  { rw [←finset.insert_erase (nat.one_mem_proper_divisors_iff_one_lt.2
+    (lt_of_lt_of_le one_lt_two hk)), finset.prod_insert (finset.not_mem_erase 1 _),
+    cyclotomic_one R],
+    have hleq : ∀ j ∈ k.proper_divisors.erase 1, 2 ≤ j,
+    { intros j hj,
+      apply nat.succ_le_of_lt,
+      exact (ne.le_iff_lt ((finset.mem_erase.1 hj).1).symm).mp (nat.succ_le_of_lt
+      (nat.pos_of_mem_proper_divisors (finset.mem_erase.1 hj).2)) },
+    have hcongr : ∀ j ∈ k.proper_divisors.erase 1, (cyclotomic j R).coeff 0 = 1,
+    { intros j hj,
+      exact hi j (nat.mem_proper_divisors.1 (finset.mem_erase.1 hj).2).2 (hleq j hj) },
+    have hrw : ∏ (x : ℕ) in k.proper_divisors.erase 1, (cyclotomic x R).coeff 0 = 1,
+    { conv_lhs {
+        apply_congr,
+        skip,
+        simp [hcongr, H] },
+      simp only [finset.prod_const_one] },
```

Comment:
```suggestion
    { rw [finset.prod_congr rfl hcongr, finset.prod_const_one] },
```
Probably one should avoid mentioning automatically-generated variable names like `H`. The lemma `finset.prod_congr` was what you were missing.

Reply:
Done it with `finset.prod_congr`.  By the way I took `simp [hcongr, H]` from the official documentation [here](https://leanprover-community.github.io/mathlib_docs/tactics.html#apply_congr). If it is better to avoid `H` maybe the documentation should modified too.